src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/iptables_linux.py

```diff
@@ -0,0 +1,159 @@
+from middlewared.service import Service, accepts, job, CallError
+
+import subprocess
+
+
+V4_FILE = '/data/v4-fw.rules'
+V6_FILE = '/data/v6-fw.rules'
+JOB_LOCK = 'firewall_rules_update'
+
+
+class IptablesService(Service):
+
+    class Config:
+        namespace = 'failover.firewall'
+        private = True
+
+    async def generate_rules(self, data):
+        """
+        Generate a list of iptables and ip6tables rules.
+
+        NOTE:
+            SSH and webUI traffic is always allowed.
+        """
+
+        # rules always start with this line
+        v4rules = ['*filter\n']
+        v6rules = ['*filter\n']
+
+        # the positions of these are important
+        # v4 default section
+        v4rules.insert(1, ':INPUT ACCEPT [0:0]\n')
+        v4rules.insert(2, ':FORWARD ACCEPT [0:0]\n')
+        v4rules.insert(3, ':OUTPUT ACCEPT [0:0]\n')
+
+        # v6 default section
+        v6rules.insert(1, ':INPUT ACCEPT [0:0]\n')
+        v6rules.insert(2, ':FORWARD ACCEPT [0:0]\n')
+        v6rules.insert(3, ':OUTPUT ACCEPT [0:0]\n')
+
+        if data['drop']:
+            # we always allow ssh and webUI access when limiting inbound
+            # connections (backwards compatibility with freeBSD HA)
+            sshport = (await self.middleware.call('ssh.config'))['tcpport']
+            web = await self.middleware.call('system.general.config')
+
+            # v4 ssh/webUI rules
+            v4rules.append(f'-A INPUT -p tcp -m tcp --dport {sshport} -j ACCEPT\n')
+            v4rules.append(f'-A INPUT -p tcp -m tcp --dport {web["ui_port"]} -j ACCEPT\n')
+            v4rules.append(f'-A INPUT -p tcp -m tcp --dport {web["ui_httpsport"]} -j ACCEPT\n')
+
+            # v6 ssh/webUI rules
+            v6rules.append(f'-A INPUT -p tcp -m tcp --dport {sshport} -j ACCEPT\n')
+            v6rules.append(f'-A INPUT -p tcp -m tcp --dport {web["ui_port"]} -j ACCEPT\n')
+            v6rules.append(f'-A INPUT -p tcp -m tcp --dport {web["ui_httpsport"]} -j ACCEPT\n')
```

Comment:
Up to this point v4 and v6 rules are exactly the same, perhaps we can come up with a way to avoid duplicating that code?